### PR TITLE
Subpass input without format

### PR DESCRIPTION
**Pull request by timvpGoogle**
This CL adds the test described in:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3060#issuecomment-887179334

### CODE REVIEW

**Comment by djdeath** [comment]
This is passing on Intel/Mesa.

File: tests/vklayertests_pipeline_shader.cpp [hunk 1]
```diff
@@ -11482,3 +11483,110 @@ TEST_F(VkLayerTest, ValidateTessellationShaderEnabled) {
...
+    // It appears it would be difficult to hit this particular case without also hitting other VUIDs first
+    m_errorMonitor->SetUnexpectedError("VUID-VkShaderModuleCreateInfo-pCode-01091");
+    m_errorMonitor->SetUnexpectedError(kVUID_Features_shaderStorageImageReadWithoutFormat_NonReadable);
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, kVUID_Features_shaderStorageImageReadWithoutFormat);
+
+    SetTargetApiVersion(VK_API_VERSION_1_2);
+    ASSERT_NO_FATAL_FAILURE(InitFramework());
+
+    if (DeviceValidationVersion() < VK_API_VERSION_1_2) {
+        printf("%s test requires Vulkan 1.2+, skipping test\n", kSkipPrefix);
```
Comment (arno-lunarg) [inline tests/vklayertests_pipeline_shader.cpp:11499]: Nit: GTEST_SKIP this?

File: tests/vklayertests_pipeline_shader.cpp [hunk 2]
```diff
@@ -11482,3 +11483,110 @@ TEST_F(VkLayerTest, ValidateTessellationShaderEnabled) {
...
+    VkAttachmentReference input = {
+        1,
+        VK_IMAGE_LAYOUT_GENERAL,
+    };
+
+    VkSubpassDescription sd = {0, VK_PIPELINE_BIND_POINT_GRAPHICS, 1, &input, 1, &color, nullptr, nullptr, 0, nullptr};
+
+    VkRenderPassCreateInfo rpci = {VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO, nullptr, 0, 2, descs, 1, &sd, 0, nullptr};
+    VkRenderPass rp;
+    VkResult err = vk::CreateRenderPass(m_device->device(), &rpci, nullptr, &rp);
```
Comment (arno-lunarg) [inline tests/vklayertests_pipeline_shader.cpp:11584]: Nit: maybe use `vk_testing::RenderPass` instead?

**Review by arno-lunarg — APPROVED** [review]
LGTM! Just 2 nits

**Comment by spencer-lunarg** [comment]
@timvpGoogle sorry this went untouched for a while, I created https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4794 and fixed up some other tests around it, tested it locally on Mesa. Thanks again for the original test

**Comment by timvpGoogle** [comment]
Sounds good. Thanks for following up and getting it cleaned up!

**Pull request closed without merging**